daemon/container.go

```diff
@@ -1052,7 +1052,7 @@ func (container *Container) setupContainerDns() error {
 		return err
 	}
 
-	if config.NetworkMode != "host" {
+	if !config.NetworkMode.IsHost() || !config.NetworkMode.IsContainer() {
```

Comment:
This one is hurting my head.

Would it be better to have a `if config.NetworkMode.IsBridge() {`, and then it's only one check, and it's a positive check?

If/when we have something more complicated than that we can always adjust (not that as it currently is wouldn't need to be adjusted anyway).